libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/PlannedRouteRefreshControllerTest.kt

```diff
@@ -0,0 +1,645 @@
+        val routes = listOf(route1, route2)
+        every {
+            RouteRefreshValidator.validateRoute(any())
+        } returns RouteRefreshValidator.RouteValidationResult.Valid
+        every { retryStrategy.shouldRetry() } returns false
+
+        sut.startRoutesRefreshing(routes)
+        val result = RouteRefresherResult(
+            false,
+            listOf(newRoute1, newRoute2),
+            RouteProgressData(1, 2, 3)
+        )
+        finishRequest(result)
+
+        verify(exactly = 1) {
+            stateHolder.onFailure(null)
+            listener.onRoutesRefreshed(result)
+        }
+        verify(exactly = 0) {
+            cancellableHandler.postDelayed(any(), any(), any())
+        }
+    }
+
+    private fun startRequest() {
+        val attemptBlocks = mutableListOf<Runnable>()
+        verify(exactly = 1) {
+            cancellableHandler.postDelayed(
+                any(),
+                capture(attemptBlocks),
+                any()
+            )
+        }
+        attemptBlocks.last().run()
+        val progressCallbacks = mutableListOf<RouteRefresherProgressCallback>()
+        verify(exactly = 1) { executor.postRoutesToRefresh(any(), capture(progressCallbacks)) }
+        progressCallbacks.last().onStarted()
+    }
+
+    private fun finishRequest(result: RouteRefresherResult) {
+        val attemptBlocks = mutableListOf<Runnable>()
+        verify(exactly = 1) {
+            cancellableHandler.postDelayed(
+                any(),
+                capture(attemptBlocks),
+                any()
+            )
+        }
+        clearMocks(cancellableHandler, answers = false)
+        attemptBlocks.last().run()
+        val progressCallbacks = mutableListOf<RouteRefresherProgressCallback>()
+        verify(exactly = 1) { executor.postRoutesToRefresh(any(), capture(progressCallbacks)) }
+        clearMocks(executor, answers = false)
+        progressCallbacks.last().onResult(result)
+    }
+
+    @Test
+    fun pause() {
+        sut.pause()
+
+        verify(exactly = 1) { cancellableHandler.cancelAll() }
+    }
+
+    @Test
+    fun resumeNoRoutes() {
+        sut.resume()
+
+        verify(exactly = 0) {
+            retryStrategy.shouldRetry()
+            cancellableHandler.postDelayed(any(), any(), any())
+        }
+    }
+
+    @Test
+    fun resumeHasRoutesShouldNotRetry() {
+        val route1 = mockk<NavigationRoute>(relaxed = true)
+        val route2 = mockk<NavigationRoute>(relaxed = true)
+        val routes = listOf(route1, route2)
+        every {
+            RouteRefreshValidator.validateRoute(any())
+        } returns RouteRefreshValidator.RouteValidationResult.Valid
+        sut.startRoutesRefreshing(routes)
+        clearAllMocks(answers = false)
+        every { retryStrategy.shouldRetry() } returns false
+
+        sut.resume()
+
+        verify(exactly = 0) { cancellableHandler.postDelayed(any(), any(), any()) }
+    }
+
+    @Test
+    fun resumeHasRoutesShouldRetry() {
```

Comment:
If some class called `PlannedRouteRefreshController#resume` without first calling `PlannedRouteRefreshController#pause` we'd have multiple refreshes queued, and they would not respect the general refresh interval. Could we make this more robust so that it's not possible to misuse?